### PR TITLE
snips-sh: init at 0.4.0

### PR DESCRIPTION
**Pull request by JeremiahSecrist**
## Description of changes
Adds a package known as snips.sh. A tui centric code snipit sharing tool. https://snips.sh/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JeremiahSecrist** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>snips-sh</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -25756,6 +25756,8 @@ with pkgs;
     php = php81;
   };
 
+  snips-sh = callPackage ../by-name/sn/snips-sh/package.nix { withTensorflow = false; };
```
Comment (Aleksanaa) [inline pkgs/top-level/all-packages.nix:25759]: You don't need this, it's already by default

File: pkgs/by-name/sn/snips-sh/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  buildInputs = [ sqlite ] ++ (lib.optional withTensorflow libtensorflow);
+
+  meta = with lib; {
+    description = "passwordless, anonymous SSH-powered pastebin with a human-friendly TUI and web UI";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    homepage = "https://snips.sh";
+    maintainers = with maintainers; [ jeremiahs ];
+    mainProgram = "snips.sh";
```
Comment (Aleksanaa) [inline pkgs/by-name/sn/snips-sh/package.nix:31]:
```suggestion
  meta = {
    description = "passwordless, anonymous SSH-powered pastebin with a human-friendly TUI and web UI";
    license = lib.licenses.mit;
    platforms = lib.platforms.linux;
    homepage = "https://snips.sh";
    maintainers = with lib.maintainers; [ jeremiahs ];
    mainProgram = "snips.sh";
```

Not required but better to

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (JeremiahSecrist) [reply pkgs/top-level/all-packages.nix:25759]: The vet script fails if otherwise

Comment (JeremiahSecrist) [reply pkgs/top-level/all-packages.nix:25759]: The new vet script would fail me otherwise.

Comment (JeremiahSecrist) [reply pkgs/top-level/all-packages.nix:25759]: So while it may not be required the script doesn't understabd that.

Comment (infinisil) [reply pkgs/top-level/all-packages.nix:25759]: I believe you might have misunderstood the CI output: You can remove this line here entirely, no change to `all-packages.nix` necessary. Can you try that?

Comment (JeremiahSecrist) [reply pkgs/top-level/all-packages.nix]: I didn't notice it's no longer required :smile:

**Comment by JeremiahSecrist** [comment]
Super excited we can have packages automatically brought in via the file structure now.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**